src/main/java/org/craftercms/engine/graphql/impl/fetchers/ContentTypeBasedDataFetcher.java

```diff
@@ -205,41 +206,72 @@ protected void processSelectedField(SelectedField currentField, BoolQueryBuilder
-                    case ARG_NAME_LT:
-                        query.filter(QueryBuilders.rangeQuery(path).lt(value));
-                        break;
-                    case ARG_NAME_GT:
-                        query.filter(QueryBuilders.rangeQuery(path).gt(value));
-                        break;
-                    case ARG_NAME_LTE:
-                        query.filter(QueryBuilders.rangeQuery(path).lte(value));
-                        break;
-                    case ARG_NAME_GTE:
-                        query.filter(QueryBuilders.rangeQuery(path).gte(value));
-                        break;
-                    case ARG_NAME_EXISTS:
-                        boolean exists = value instanceof Boolean ? (Boolean)value : Boolean.parseBoolean(value.toString());
-                        if (exists) {
-                            query.filter(QueryBuilders.existsQuery(path));
-                        } else {
-                            query.mustNot(QueryBuilders.existsQuery(path));
-                        }
-                        break;
-                    default:
-                        // never happens
+            filters.forEach((name, value) -> addFieldFilter(name, path, value, query));
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    protected void addFieldFilter(String name, String path, Object value, BoolQueryBuilder query) {
+        if (value instanceof List) {
+            List<Map<String, Object>> actualFilters = (List<Map<String, Object>>) value;
+            switch (name) {
+                case ARG_NAME_NOT:
+                    BoolQueryBuilder notQuery = QueryBuilders.boolQuery();
+                    actualFilters.forEach(notFilter -> notFilter.forEach((notName, notValue) ->
+                        addFieldFilter(notName, path, notValue, notQuery)));
+                    query.mustNot(notQuery);
+                    break;
+                case ARG_NAME_AND:
+                    BoolQueryBuilder andQuery = QueryBuilders.boolQuery();
+                    actualFilters.forEach(orFilter -> orFilter.forEach((orName, orValue) ->
+                        addFieldFilter(orName, path, orValue, andQuery)));
+                    query.filter(andQuery);
+                    break;
+                case ARG_NAME_OR:
+                    BoolQueryBuilder orQuery = QueryBuilders.boolQuery();
+                    actualFilters.forEach(orFilter -> {
+                        BoolQueryBuilder shouldQuery = QueryBuilders.boolQuery();
+                        orFilter.forEach((orName, orValue) -> addFieldFilter(orName, path, orValue, shouldQuery));
+                        orQuery.filter(shouldQuery);
+                    });
+                    BoolQueryBuilder realOrQuery = QueryBuilders.boolQuery();
+                    orQuery.filter().forEach(realOrQuery::should);
+                    query.must(realOrQuery);
+                    break;
+                default:
+                    // never happens
```

Comment:
what if they send the wrong filter, something unrecognized? shouldn't we be throwing an exception?

Reply:
GraphQL will never allow to execute an unrecognized filter, only those defined in the schema